### PR TITLE
Make Max and Min functions and remove Dense methods

### PR DESCRIPTION
**Pull request by btracey**
Updates #38

### CODE REVIEW

**Comment by btracey** [comment]
Fixing min.

File: mat64/matrix.go [hunk 1]
```diff
@@ -351,6 +352,171 @@ func Inverse(a Matrix) (*Dense, error) {
...
+		}
+		if rm.Uplo == blas.Upper {
+			max := 0.0
+			for i := 0; i < rm.N; i++ {
+				for _, v := range rm.Data[i*rm.Stride+i : i*rm.Stride+rm.N] {
+					if v > max {
+						max = v
+					}
+				}
+			}
```
Comment (vladimir-ch) [inline mat64/matrix.go]: Missing return

File: mat64/list_test.go [hunk 2]
```diff
@@ -58,8 +63,8 @@ func sameAnswerFloat(a, b interface{}) bool {
 	return a.(float64) == b.(float64)
 }
 
-// legalTypeAll returns true for all Matrix types.
-func isAny(a Matrix) bool {
+// isAnyType returns true for all Matrix types.
+func isAnyType(a Matrix) bool {
```
Comment (kortschak) [inline mat64/list_test.go]: s/a//

File: mat64/matrix.go [hunk 3]
```diff
@@ -351,6 +352,171 @@ func Inverse(a Matrix) (*Dense, error) {
...
+		}
+		return max
+	case RawTriangular:
+		rm := m.RawTriangular()
+		// The max of a triangular is at least 0 unless the size is 1.
+		if rm.N == 1 {
+			return rm.Data[0]
+		}
+		if rm.Uplo == blas.Upper {
+			max := 0.0
```
Comment (kortschak) [inline mat64/matrix.go]: Move this one line up and delete the decl below. Same for min.

**Comment by kortschak** [comment]
LGTM with minor comments - and return.

**Comment by kortschak** [comment]
Updates #138.

**Pull request merged**